### PR TITLE
[Refactor] UserType in Notification Classes

### PR DESCRIPTION
**Pull request by johnxnguyen**
## What's new in this PR?

This PR is part of an ongoing effort to all public references that leak concrete user objects. See the architecture issues for more info: https://github.com/wireapp/ios-architecture/issues/89

Public exposure of `ZMUser` has been removed from the following cluster of files:

- `NotificationUserInfo.swift`
- `ZMLocalNotification.swift`
- `LocalNotificationType+Localization.swift`
- `LocalNotificationDispatcher+Calling.swift`

## Note

I came across a lot of unnecessary public access modifiers, so I've removed those too to harden the SE boundary.

### CODE REVIEW

File: Source/Notifications/Push notifications/LocalNotificationDispatcher+Calling.swift [hunk 1]
```diff
@@ -20,25 +20,26 @@ import Foundation
 
 public extension LocalNotificationDispatcher {
     
-    func process(callState: CallState, in conversation: ZMConversation, caller: ZMUser) {
-        // missed call notification are handled separately
-        // but if call was answered elsewhere then proceed
+    func process(callState: CallState, in conversation: ZMConversation, caller: UserType) {
```
Comment (typfel) [inline Source/Notifications/Push notifications/LocalNotificationDispatcher+Calling.swift]: Isn't this an internal where it's ok to use the concrete `ZMUser` type

File: Source/Notifications/Push notifications/LocalNotificationDispatcher+Calling.swift [hunk 2]
```diff
@@ -20,25 +20,26 @@ import Foundation
...
+
+        guard let caller = caller as? ZMUser else { return }
         let note = ZMLocalNotification(callState: callState, conversation: conversation, caller: caller)
         callingNotifications.cancelNotifications(conversation)
         note.apply(scheduleLocalNotification)
         note.apply(callingNotifications.addObject)
     }
     
-    func processMissedCall(in conversation: ZMConversation, caller: ZMUser) {
+    func processMissedCall(in conversation: ZMConversation, caller: UserType) {
```
Comment (typfel) [inline Source/Notifications/Push notifications/LocalNotificationDispatcher+Calling.swift]: Also internal?

**Comment by johnxnguyen** [comment]
@typfel I've reverted the commits you reviewed and instead removed public access. I also removed a lot of redundant public access in related classes (but not strictly necessary for the `ZMUser` type leakage.

**Comment by johnxnguyen** [comment]
@zenkins release major

**Pull request merged**